diff -rau lilo-23.2//Makefile lilo-23.2.patched//Makefile
--- lilo-23.2//Makefile	2011-04-09 17:05:59.000000000 +0000
+++ lilo-23.2.patched//Makefile	2011-05-04 10:04:14.356600225 +0000
@@ -38,14 +38,14 @@
 #
 all: test
 	$(MAKE) -C src all
-	$(MAKE) -C images all
+#	$(MAKE) -C images all
 
 #
 # everything above plus the statically linked version
 #
 alles: test
 	$(MAKE) -C src alles
-	$(MAKE) -C images all
+#	$(MAKE) -C images all
 
 #
 # documentation files
Only in lilo-23.2.patched/: Makefile~
diff -rau lilo-23.2//man/mkrescue.8 lilo-23.2.patched//man/mkrescue.8
--- lilo-23.2//man/mkrescue.8	2011-04-09 17:05:59.000000000 +0000
+++ lilo-23.2.patched//man/mkrescue.8	2011-05-04 09:36:51.486471416 +0000
@@ -149,7 +149,7 @@
 .BR lilo (8),
 .BR mkfs (8),
 .BR mkinitrd (8),
-.BR mkisofs (8),
+.BR genisoimage (8),
 .BR mount (8)
 
 .\" .SH "AUTHOR"
Only in lilo-23.2.patched//man: mkrescue.8~
diff -rau lilo-23.2//mkrescue lilo-23.2.patched//mkrescue
--- lilo-23.2//mkrescue	2011-04-09 17:05:59.000000000 +0000
+++ lilo-23.2.patched//mkrescue	2011-05-04 09:38:44.306067934 +0000
@@ -683,7 +683,7 @@
 	    opt=-hard-disk-boot
 	fi
 	mv $mfile $mount/boot.bin
-	mkisofs $opt -J -R -T \
+	genisoimage $opt -J -R -T \
 		-V LILO_BOOT -A "Linux Boot CD created by LILO mkrescue" \
 		-b boot.bin -c boot.cat -o $out $mount
 	cat <<EOF
@@ -691,10 +691,10 @@
 END MKISOFS:  output is in  '$device'
 
 
-The bootable CD can be burned with the 'cdrecord' utility
+The bootable CD can be burned with the 'wodim' utility
 using a command of the form:
 
-	cdrecord [<options>] [dev=<device>] $device
+	wodim [<options>] [dev=<device>] $device
 
 EOF
 elif [ "$fast" = fast ]; then
Only in lilo-23.2.patched/: mkrescue~
diff -rau lilo-23.2//src/boot.c lilo-23.2.patched//src/boot.c
--- lilo-23.2//src/boot.c	2011-04-09 17:05:59.000000000 +0000
+++ lilo-23.2.patched//src/boot.c	2011-05-04 09:41:18.316315253 +0000
@@ -328,6 +328,7 @@
 #endif
     struct stat st;
     char *pos;
+    char *vanilla_part = part;
     int i, code;
     int letter = 0;
     int unsafe;
@@ -382,6 +383,7 @@
     if (unsafe) {
 	(void) geo_open_boot(&geo,boot);
 	if (part) die("TABLE and UNSAFE are mutually incompatible.");
+	if (vanilla_part) die("TABLE and UNSAFE are mutually incompatible.");
     }
     else {
 	b_fd = geo_open(&geo,boot,O_RDONLY);
@@ -405,7 +407,8 @@
 	    if (size < 0) die("read %s: %s",boot,strerror(errno));
 	    else die("Can't get magic number of %s",boot); }
 	if (magic != BOOT_SIGNATURE)
-	    die("First sector of %s doesn't have a valid boot signature",boot);
+           fprintf(errstd,"Warning: First sector of %s doesn't have a valid boot signature\n",boot);
+
     }
 
 /* process the 'master-boot' or 'boot-as' options */
Only in lilo-23.2.patched//src: boot.c~
diff -rau lilo-23.2//src/bsect.c lilo-23.2.patched//src/bsect.c
--- lilo-23.2//src/bsect.c	2011-04-09 17:05:59.000000000 +0000
+++ lilo-23.2.patched//src/bsect.c	2011-05-04 09:44:52.066315117 +0000
@@ -499,7 +499,7 @@
         nowarn = 0;
 
 	warn("'-F' override used. Filesystem on  %s  may be destroyed.", boot_devnam);
-	if (!yesno("\nProceed? ",0)) exit(0);
+	if (!yesno("\nProceed? ",0)) exit(1);
 
         nowarn = nowarn2;
 	ireloc=PTW_OKAY;
@@ -1452,6 +1452,7 @@
 #if 1
     if (ireloc &&
     	  bsect.par_1.cli == 0xFA
+	  && !has_partitions(boot_dev_nr)
     	 						 ) {
 /* perform the relocation of the boot sector */
 	int len = bsect.par_1.code_length;
Only in lilo-23.2.patched//src: bsect.c~
diff -rau lilo-23.2//src/crt.S lilo-23.2.patched//src/crt.S
--- lilo-23.2//src/crt.S	2011-04-09 17:05:59.000000000 +0000
+++ lilo-23.2.patched//src/crt.S	2011-05-04 09:48:46.446157787 +0000
@@ -11,7 +11,6 @@
 
 XPITCH  =	MAX_IMAGE_NAME+6
 MINROW	=	4
-MAX2COL	=	14
 MAXTITLE  =	3
 MAXCOMMENT  =	3
 
@@ -77,12 +76,8 @@
 
 	mov	ax,si
 	mov	bl,#2		;set for 2 columns
-	cmp	al,#MAX2COL
-	jle	col1
-	inc	bl
-col1:	mov	[ncol],bl
-	add	al,bl
-	dec	al
+	mov	[ncol],bl
+	inc	al
 	div	bl
 	cmp	al,#MINROW
 	jg	row1
diff -rau lilo-23.2//src/device.c lilo-23.2.patched//src/device.c
--- lilo-23.2//src/device.c	2011-04-09 17:05:59.000000000 +0000
+++ lilo-23.2.patched//src/device.c	2011-05-04 09:45:17.276315478 +0000
@@ -854,7 +854,7 @@
 
     if (ret && fatal) {
 	fprintf(stderr, "Aborting ...\n");
-	exit(0);
+	exit(1);
     }
     
     return ret;
@@ -1013,7 +1013,7 @@
 		DEVICE Dev2;
 		serial = volid_get_set(dev, 0, ID_GET);
 
-#if 0
+#if 1
 /* take care of uninitialized Volume IDs with no fanfare */
 		if (serial==0) {
 		    serial = volid_get_set(dev, new_serial(dev), ID_SET);
Only in lilo-23.2.patched//src: device.c~
diff -rau lilo-23.2//src/edit.c lilo-23.2.patched//src/edit.c
--- lilo-23.2//src/edit.c	2011-04-09 17:05:59.000000000 +0000
+++ lilo-23.2.patched//src/edit.c	2011-05-04 09:46:10.146624951 +0000
@@ -688,7 +688,7 @@
     }
 
     printf("Transfer parameters from '%s' to '%s'", config_file, bitmap_file);
-    if (yesno("?", 0)==0) exit(0);
+    if (yesno("?", 0)==0) exit(1);
 
     if (verbose > 0) printf("%s bitmap file:  %s\n", opt, bitmap_file);
     
Only in lilo-23.2.patched//src: edit.c~
diff -rau lilo-23.2//src/lilo.h lilo-23.2.patched//src/lilo.h
--- lilo-23.2//src/lilo.h	2011-04-09 17:05:59.000000000 +0000
+++ lilo-23.2.patched//src/lilo.h	2011-05-04 09:33:53.276156247 +0000
@@ -152,7 +152,7 @@
 #define MAJOR_MDP	254 /* Enhanced multi-disk RAID sets [experimental?] */
 
 #define MAX_TOKEN	1023 /* max device Token length */
-#define MAX_IMAGE_NAME	15 /* maximum name length (w/o terminating NUL) */
+#define MAX_IMAGE_NAME	31 /* maximum name length (w/o terminating NUL) */
 #define MAX_DESCR_SECTORS  12  /* upper limit on MAX_DESCR_SECS */
 
 #ifdef LCF_PASS160
Only in lilo-23.2.patched//src: lilo.h~
diff -rau lilo-23.2//src/partition.c lilo-23.2.patched//src/partition.c
--- lilo-23.2//src/partition.c	2011-04-09 17:05:59.000000000 +0000
+++ lilo-23.2.patched//src/partition.c	2011-05-04 09:46:25.796625107 +0000
@@ -235,7 +235,7 @@
 	if (!dos && !cfg_get_flag(cf_options,"ignore-table"))
 	    die("You may proceed by using either '-P ignore' or 'ignore-table'");
 #else
-	if (!yesno("\nProceed? ", 0)) exit(0);
+	if (!yesno("\nProceed? ", 0)) exit(1);
 #endif
     }
     cyl = part_table[part].cyl+((part_table[part].sector >> 6) << 8);
Only in lilo-23.2.patched//src: partition.c~
diff -rau lilo-23.2//src/second.S lilo-23.2.patched//src/second.S
--- lilo-23.2//src/second.S	2011-04-09 17:05:59.000000000 +0000
+++ lilo-23.2.patched//src/second.S	2011-05-04 09:48:46.446157787 +0000
@@ -622,7 +622,7 @@
 	call	say
 	add	si,#MAX_IMAGE_NAME+4
 	inc	dl		; count the image
-	test	dl,#3		; inside line -> go on
+	test	dl,#1		; inside line -> go on
 	jnz	fill
 	call	crlf
 	jmp	imgdne		; next image
@@ -635,7 +635,7 @@
 	jbe	fill
 imgdne:	add	si,#id_size-MAX_IMAGE_NAME-4
 	loop	lloop		; next image
-ldone:	test	dl,#3		; already at BOL ?
+ldone:	test	dl,#1		; already at BOL ?
 	jz	atbol		; yes -> no CRLF
 	call	crlf
 atbol:	
@@ -2927,7 +2927,8 @@
 	xor	edx,edx			; flag nothing found
 	xor	esi,esi			; flag size==0
 	xor	ebx,ebx
-	jmp	e8go
+;	jmp	e8go			; Circumvented due to BIOS problems
+	br      no_e801			; Skip to backup plan
 e8go2:	or	ebx,ebx			; test for end
 	jz	e8go5
 e8go:	push	edx			; save best prospect
Only in lilo-23.2.patched//src: second.S~
